lib/utils/background_push.dart

```diff
@@ -296,9 +297,43 @@ class BackgroundPush {
     );
   }
 
-  Future<void> goToRoom(NotificationResponse? response) async {
+  Future<void> getIosInitialNoti() async {
+    try {
+      final noti = await apnChannel.invokeMethod('getInitialNoti');
+      Logs().v('[Push] Got initial notification: $noti');
+      if (noti != null) {
+        iOSUserSelectedNoti(noti);
+      }
+    } catch (e, s) {
+      Logs().e('[Push] Failed to get initial notification', e, s);
+    }
+  }
+
+  void iOSUserSelectedNoti(dynamic noti) {
+    // roomId is payload if noti is local
+    final roomId = noti['room_id'] ?? noti['payload'];
+    goToRoom(roomId);
+  }
+
+  void onReceiveNotification(dynamic message) {
+    Logs().d('BackgroundPush::onMessage(): $message');
+    final notification = _parseMessagePayload(message);
+    pushHelper(
+      notification,
+      client: client,
+      l10n: l10n,
+      activeRoomId: router?.currentState?.pathParameters['roomid'],
+      onSelectNotification: onSelectNotification,
+    );
+    Logs().d('BackgroundPush::onMessage(): finished pushHelper');
```

Comment:
perhaps you could log when pushHelper is done with `whenComplete` ?

Reply:
@imGok  I just move the old listener handler to function to reuse